pandas/tests/io/formats/test_to_csv.py

```diff
@@ -514,3 +514,85 @@ def test_to_csv_compression(self, compression_only, read_infer, to_infer):
             df.to_csv(path, compression=to_compression)
             result = pd.read_csv(path, index_col=0, compression=read_compression)
             tm.assert_frame_equal(result, df)
+
+    def test_to_csv_string_with_sjis(self):
+        data = {"int": [1, 2, 3], "str_sjis": ["abc", "\u070a", "def"]}
+        df = pd.DataFrame(data)
+        # case 1: errors=strict
+        with tm.ensure_clean("sjis_test.csv") as path:
+            expected_exception = UnicodeEncodeError
+            with pytest.raises(expected_exception):
+                df.to_csv(
+                    path,
+                    line_terminator="\n",
+                    encoding="sjis",
+                    errors="strict",
+                    index=False,
+                )
+
+        # case 2: errors=replace
+        with tm.ensure_clean("sjis_test.csv") as path:
+            expected_sjis = b"int,str_sjis\n" b"1,abc\n" b"2,?\n" b"3,def\n"
```

Comment:
Do you need multiple strings here on this line?

Reply:
Do you mean `expected_sjis = "int,str_sjis\n1,abc\n2,?\n3,def\n"` ?

Reply:
Sorry, you mean

```
            expected_cp932 = """\
int,str_cp932
1,abc
2,?
3,def
"""
```

Reply:
Either of those. Just shouldn't need the `b"ab\n"b"cd\n"b"ef\n".